tests/integration/test_load_and_serve_agent.py

```
@@ -0,0 +1,185 @@
+            else:
+                traces[None] = trace
+"""
+class SpanNode:
+    def __init__(span):
+        _span: ReadableSpan = span
+        _children: list["SpanNode"] = []
+    def add_span(child_span):
+        """Add child span sorted according to timestamps"""
+        ...
+
+def check_uvx_installed() -> bool:
+    """The integration tests requires uvx"""
+    try:
+        result = subprocess.run(  # noqa: S603
+            ["uvx", "--version"],  # noqa: S607
+            capture_output=True,
+            check=True,
+        )
+        return True if result.returncode == 0 else False  # noqa: TRY300
+    except Exception:
+        return False
+
+
+@pytest.mark.skipif(
+    os.environ.get("ANY_AGENT_INTEGRATION_TESTS", "FALSE").upper() != "TRUE",
+    reason="Integration tests require `ANY_AGENT_INTEGRATION_TESTS=TRUE` env var",
+)
+@pytest.mark.skipif(
+    not any_agent.serving.serving_available,
+    reason="Integration tests require the installation of the ADK samples (`pip install 'git+https://github.com/google/A2A#subdirectory=samples/python'`)",
+)
+@pytest.mark.asyncio
+async def test_load_and_serve_agent(
+    agent_framework: AgentFramework, tmp_path: Path
+) -> None:
+    kwargs = {}
+
+    tmp_file = "tmp.txt"
+    logger.info("Starting")
+
+    if not check_uvx_installed():
+        msg = "uvx is not installed. Please install it to run this test."
+        raise RuntimeError(msg)
+
+    def write_file(text: str) -> None:
+        """write the text to a file in the tmp_path directory
+
+        Args:
+            text (str): The text to write to the file.
+
+        Returns:
+            None
+        """
+        with open(os.path.join(tmp_path, tmp_file), "w", encoding="utf-8") as f:
+            f.write(text)
+
+    kwargs["model_id"] = "gpt-4.1-mini"
+    env_check = validate_environment(kwargs["model_id"])
+    if not env_check["keys_in_environment"]:
+        pytest.skip(f"{env_check['missing_keys']} needed for {agent_framework}")
```

Should we raise an error instead of skipping? idk, I think this logic exists in other tests too

---

I didn't change this... :-/ I'm ok to do whatever is currently done in the rest of the tests if the policy has changed since the start of this PR.

---

I didn't change this AFAICT 😅

---

This is from #167